src/platform/plugins/shared/unified_search/public/query_string_input/query_string_input.tsx

```diff
@@ -732,6 +739,20 @@ export class QueryStringInput extends PureComponent<QueryStringInputProps, State
     }
   });
 
+  handleBlurOnScroll = onRaf(() => {
+    // for small screens, unified search bar is no longer sticky,
+    // so we need to blur the input when it scrolls out of view
+    const isSmallScreen = window.innerWidth < 768;
```

Comment:
I don't know if `euiTheme.breakpoint.m` evaluates to 768, but we might want to consider using it too :)